clients/input_copy.go

```diff
@@ -197,12 +218,25 @@ func CopyFile(ctx context.Context, sourceURL, destOSBaseURL, filename, requestID
 		byteAccWriter := ByteAccumulatorWriter{count: 0}
 		defer func() { writtenBytes = byteAccWriter.count }()
 
+		var c io.ReadCloser
 		c, err := getFile(ctx, requestID, sourceURL, dStorage)
+
 		if err != nil {
-			return fmt.Errorf("download error: %w", err)
+			glog.Errorf("error getting file: %w", err)
+			return err
 		}
+
 		defer c.Close()
 
+		if decryptor != nil {
+			decryptedFile, err := crypto.DecryptAESCBC(c, &decryptor.decryptKey, decryptor.encryptedKey)
```

Comment:
Going even further, maybe `DecryptionKeys` should be in the `crypto` package and be the args for this func.